eth/vm/base.py

```diff
@@ -576,10 +583,12 @@ def validate_block(self, block: BlockAPI) -> None:
             )
 
         if not self.chaindb.exists(block.header.state_root):
-            raise ValidationError(
-                "`state_root` was not found in the db.\n"
-                f"- state_root: {block.header.state_root!r}"
-            )
+            if not self.state.make_state_root() == block.header.state_root:
```

Comment:
This can be cleaned up a bit. The state root was not in the db but if we make the state root at this point in time it does match... that's what this is checking for.